File: src/csharp/Intel/Iced.UnitTests/Intel/FormatterTests/SymbolResolverTestUtils.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if GAS || INTEL || MASM || NASM || FAST_FMT
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace Iced.UnitTests.Intel.FormatterTests {
	static class SymbolResolverTestUtils {
		public static IEnumerable<object[]> GetFormatData(string formatterDir, string formattedStringsFile) {
			var (tests, ignored) = SymbolResolverTestCases.AllTests;
			var formattedStrings = FileUtils.ReadRawStrings(Path.Combine(formatterDir, formattedStringsFile)).ToArray();
			formattedStrings = Utils.Filter(formattedStrings, ignored);
			if (tests.Length != formattedStrings.Length)
				throw new ArgumentException($"(tests.Length) {tests.Length} != (formattedStrings.Length) {formattedStrings.Length} . tests[0].HexBytes = {(tests.Length == 0 ? "<EMPTY>" : tests[0].HexBytes)} & formattedStrings[0] = {(formattedStrings.Length == 0 ? "<EMPTY>" : formattedStrings[0])}");
			var res = new object[tests.Length][];
			for (int i = 0; i < tests.Length; i++)
				res[i] = new object[3] { i, tests[i], formattedStrings[i] };
			return res;
		}
	}
}
#endif

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/fmt/SymbolResolverTestUtils.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.fmt;

import java.nio.file.Paths;
import java.util.ArrayList;

import org.junit.jupiter.params.provider.Arguments;

final class SymbolResolverTestUtils {
	public static ArrayList<Arguments> getFormatData(String formatterDir, String formattedStringsFile) {
		SymbolResolverTestCases.Info info = SymbolResolverTestCases.allTests;
		String[] formattedStrings = FmtFileUtils.readRawStrings(Paths.get(formatterDir, formattedStringsFile).toString()).toArray(new String[0]);
		formattedStrings = Utils.filter(formattedStrings, info.ignored);
		if (info.tests.length != formattedStrings.length)
			throw new UnsupportedOperationException();
		ArrayList<Arguments> result = new ArrayList<Arguments>(info.tests.length);
		for (int i = 0; i < info.tests.length; i++)
			result.add(Arguments.of(i, info.tests[i], formattedStrings[i]));
		return result;
	}
}
